adal/src/main/java/com/microsoft/aad/adal/IJWSBuilder.java

```diff
@@ -49,4 +53,9 @@
      */
     String generateSignedJWT(String nonce, String submitUrl, RSAPrivateKey privateKey,
                              RSAPublicKey pubKey, X509Certificate x509Certificate) throws AuthenticationException;
+
+
+    String generateJWT(HashMap<String, String> header,
```

Comment:
Can this just be a `Map`?

Reply:
since we're using HashMap for the responseHeader and responseBody, just as you mentioned as above, I think we'd better to keep it consistent to avoid any unexpected error.